tool/tctl/common/token_command.go

```diff
@@ -178,18 +183,26 @@ func (c *TokenCommand) List(client auth.ClientI) error {
 	// Sort by expire time.
 	sort.Slice(tokens, func(i, j int) bool { return tokens[i].Expiry().Unix() < tokens[j].Expiry().Unix() })
 
-	tokensView := func() string {
-		table := asciitable.MakeTable([]string{"Token", "Type", "Expiry Time (UTC)"})
-		for _, t := range tokens {
-			expiry := "never"
-			if t.Expiry().Unix() > 0 {
-				expiry = t.Expiry().Format(time.RFC822)
+	if c.format == teleport.Text {
+		tokensView := func() string {
+			table := asciitable.MakeTable([]string{"Token", "Type", "Expiry Time (UTC)"})
+			for _, t := range tokens {
+				expiry := "never"
+				if t.Expiry().Unix() > 0 {
+					expiry = t.Expiry().Format(time.RFC822)
+				}
+				table.AddRow([]string{t.GetName(), t.GetRoles().String(), expiry})
 			}
-			table.AddRow([]string{t.GetName(), t.GetRoles().String(), expiry})
+			return table.AsBuffer().String()
+		}
+		fmt.Printf(tokensView())
+	} else {
```

Comment:
Is "c.format" validated anywhere that it is either "text" or "json"? If not, this will accept anything (--format=asd) and render json instead of producing error. Same q for similar blocks below.